rust/src/ctypes.rs

```diff
@@ -0,0 +1,33 @@
+use ffi_utils::strings::{
+    string_to_c_char
+};
+
+use std;
+use std::os::raw::c_char;
+
+use errors;
+use Toodle;
+
+pub struct ResultC<T> {
+    pub value: *const T,
+    pub error: *const c_char
+}
+
+impl From<std::result::Result<Toodle, errors::Error>> for ResultC<Toodle> {
```

Comment:
I'm pretty sure we can implement a trait for a generic type, so
```
impl From<std::Result::Result<T, list_errors::Error>> for ResultC<T>
```

If we can, this would be cool as the only thing we are doing here that cares about types is the error parsing and we won't have to create multiple trait implementations.